src/google/protobuf/descriptor.proto

```diff
@@ -351,6 +351,14 @@ message FileOptions {
   // generated classes from this .proto. There is no default.
   optional string objc_class_prefix = 36;
 
+  // Namespace for generated classes; defaults to the package.
+  optional string csharp_namespace = 37;
+
+  // A nested namespace for the umbrella class.  Helpful for name collisions caused by
```

Comment:
Modified comment to be more explanatory. Btw, how happy are you with the word "umbrella"? I think it's not as explanatory as it could be.